state/txindex/indexer_service.go

```diff
@@ -119,9 +137,82 @@ func (is *IndexerService) OnStart() error {
 	return nil
 }
 
+func (is *IndexerService) Prune(retainHeight int64) {
+	lastRetainHeight, err := is.getLastIndexerRetainHeight()
+	if err != nil {
+		panic(err)
+	}
+	if retainHeight <= lastRetainHeight {
+		return
+	}
+	is.txIdxr.Prune(lastRetainHeight, retainHeight)
+	is.blockIdxr.Prune(lastRetainHeight, retainHeight)
+	is.setLastIndexerRetainHeight(retainHeight)
```

Comment:
Can it ever happen that the block and tx indexer do not prune until retain height?